src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -305,9 +310,19 @@ void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion
         break;
     }
 
-    // 3. Actually modify the selection
+    // 2.B) Clamp the movement to the mutable viewport
+    const auto bufferSize = _activeBuffer().GetSize();
+    const auto mutableViewport = _GetMutableViewport();
+    const COORD bottomRightInclusive{ mutableViewport.RightInclusive(), mutableViewport.BottomInclusive() };
+    if (bufferSize.CompareInBounds(targetPos, bottomRightInclusive) > 0)
+    {
+        targetPos = bottomRightInclusive;
+    }
+
+    // 3. Actually modify the selection state

```

Comment:
```suggestion
    // 3. Actually modify the selection
```

Bad merge with 11372